src/fmu/tools/sensitivities/design_distributions.py

```diff
@@ -487,21 +487,57 @@ def sample_discrete(dist_params, numreals):
     status = True
     outcomes = re.split(",", dist_params[0])
     outcomes = [item.strip() for item in outcomes]
+
+    if numreals == 0:
+        return status, np.array([])
+
+    if numreals < 0:
+        raise ValueError("numreal must be a positive integer")
+
     if len(dist_params) == 2:  # non uniform
         weights = re.split(",", dist_params[1])
         if len(outcomes) != len(weights):
             raise ValueError(
                 "Number of weights for discrete distribution "
                 "is not the same as number of values."
             )
-        weightnmbr = [float(weight) for weight in weights]
-        fractions = [weight / sum(weightnmbr) for weight in weightnmbr]
-        values = np.random.choice(outcomes, numreals, p=fractions)
+
+        try:
+            probabilities = [float(weight) for weight in weights]
+        except ValueError as e:
+            raise ValueError(
+                "All weights must be valid floating point numbers. "
+                f"Got weights: {weights}"
+            ) from e
+
+        # Validate probabilities
+        if not all(0 <= p <= 1 for p in probabilities):
+            raise ValueError("All probabilities must be between 0 and 1")
+        if not np.isclose(sum(probabilities), 1.0):
+            raise ValueError("Probabilities must sum to 1")
+
+        fractions = probabilities
+
     elif len(dist_params) == 1:  # uniform
-        values = np.random.choice(outcomes, numreals)
+        fractions = [1.0 / len(outcomes)] * len(outcomes)
+
     else:
         status = False
```

Comment:
Maybe the status variable can be removed as it is not really used.

Reply:
Maybe but probably not in this PR.